tests/analysis_agent.py

```diff
@@ -0,0 +1,213 @@
+
+        Returns:
+            List[Workflow]: A list of Workflow objects. E.g. [Workflow_1, ..., Workflow_100]
+
+        Raises:
+            requests.exceptions.HTTPError: When the request to Secondary-analysis service (Cromwell) failed.
+        """
+        query_dict = {
+            'label': {
+                'bundle-uuid': bundle_uuid
+            }
+        }
+
+        if with_labels:
+            query_dict['additionalQueryResultFields'] = ['labels']
+
+        if bundle_version:
+            query_dict['label']['bundle-version'] = bundle_version
+        
+        query_dict['label']['caas-collection-name'] = self.cromwell_collection
+
+        response = cwm_api.query(query_dict=query_dict, auth=self.auth)
+        response.raise_for_status()
+        result = response.json()
+        all_workflows = [Workflow(wf) for wf in result['results']]
+        return all_workflows
+
+    def query_by_project_uuid(self, project_uuid, with_labels=False):
+        """Query the analysis workflows by the HCA DCP Ingest submission project-UUID, which is essentially one of the
+            workflow labels.
+
+        Note, due to the open issue: https://github.com/broadinstitute/cromwell/issues/3115, if the result of workflows
+        are more than ~1000, this function will very likely raise an error. The `with_labels` is a flag controlling the
+        behavior of whether to query the workflows asking for the labels in the response, by default it's set to True,
+        so please set it to False if you don't want to risk getting error responses.
+
+        Args:
+            project_uuid (str): HCA DCP Ingest submission project-UUID.
+            with_labels (bool): Optional, whether to return workflow labels in the results. By default, it's False.
+
+        Returns:
+            List[Workflow]: A list of Workflow objects. E.g. [Workflow_1, ..., Workflow_100]
+
+        Raises:
+            requests.exceptions.HTTPError: When the request to Secondary-analysis service (Cromwell) failed.
+        """
+        query_dict = {
+            "label": {
+                "project_uuid": project_uuid
+            }
+        }
+        if with_labels:
+            query_dict['additionalQueryResultFields'] = ['labels']
+        
+        query_dict['label']['caas-collection-name'] = self.cromwell_collection
+
+        response = cwm_api.query(query_dict=query_dict, auth=self.auth)
+        response.raise_for_status()
+        result = response.json()
+        all_workflows = [Workflow(wf) for wf in result['results']]
+        return all_workflows
+
+    def query_by_project_shortname(self, project_shortname, with_labels=False):
+        """Query the analysis workflows by the HCA DCP Ingest submission project-shortname, which is essentially one of the
+            workflow labels.
+
+        TODO: consolidate this function with the above "project_uuid" so we can re-use a lot of code
```

Comment:
Maybe have a "query_by_label" function?

Reply:
Do you mind if I hold off this change? I still think to put the agent directly here and in dcp-diag should be a temp solution. We might need to put a polished/optimized analysis agent in somewhere like secondary-analysis or cromwell-tools and import it from any consumers(just like what the dss_agent is doing). So I prefer to keep this code simple here and don't invest a lot on polishment :octocat: